lib/logger/lib/logger.ex

```diff
@@ -515,10 +513,22 @@ defmodule Logger do
   """
   @spec disable(pid) :: :ok
   def disable(pid) when pid == self() do
-    Process.put(@metadata, {false, metadata()})
+    Process.put(@metadata, false)
     :ok
   end
 
+  @doc """
+  Returns whether the logging is enabled for given process.
+
+  Currently the only accepted PID is `self()`.
+  """
+  def enabled?(pid) when pid == self() do
+    case Process.get(@metadata) do
```

Comment:
Probably `Process.get(@metadata, true)` would be enough (without the `case`).

Reply:
Oh, I forgot that `Process.metadata` supports `default` argument.